pkg/agent/monitortool/monitor.go

```diff
@@ -357,6 +366,33 @@ func (m *NodeLatencyMonitor) pingAll(ipv4Socket, ipv6Socket net.PacketConn) {
 	klog.V(4).InfoS("Done pinging all Nodes")
 }
 
+// GetSummary returns the latency summary of the given Node IP.
+func (m *NodeLatencyMonitor) GetSummary() *statsv1alpha1.NodeLatencyStats {
+	return &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: m.nodeName,
+		},
+		PeerNodeLatencyStats: m.latencyStore.ConvertList(m.nodeName),
+	}
+}
+
+func (m *NodeLatencyMonitor) report() {
+	summary := m.GetSummary()
+	if summary == nil {
+		klog.InfoS("Latency summary is nil")
+		return
+	}
```

Comment:
This is not possible. `GetSummary` always returns a non-nil `*statsv1alpha1.NodeLatencyStats`.

Reply:
That's correct, it was a pointer that was returned in the previous implementation, I will remove it.